src/lib/builders/menu/create.ts

```diff
@@ -332,7 +332,10 @@ export function createMenuBuilder(opts: _MenuBuilderOptions) {
 						handleRovingFocus(itemEl);
 						return;
 					}
-					rootOpen.set(false);
+					// Allows forms to submit before the menu is removed from the DOM
+					sleep(1).then(() => {
+						rootOpen.set(false);
```

Comment:
I wonder why this is needed? Dialog works without this, for example. If this is needed here, I'd assume it'd be needed anywhere we use a portal.

Also, devs should simply cancel the behaviour with event listeners?

Reply:
Yeah but they don't necessarily want to cancel the behavior, they want the menu to close but they also want a form to submit. The dropdown menu works without this as well, which is strange because the context menu uses the same `item` as the dropdown, but the context doesn't,

Reply:
More specifically, this is what I get in the browser console when trying to submit with:
![image](https://github.com/melt-ui/melt-ui/assets/64506580/31b568c0-7c72-41be-a66d-da5f6b7ac479)

Skeleton had a similar issue with their stepper which is where I got the idea to try this:
https://github.com/skeletonlabs/skeleton/pull/1333/files

Reply:
Could also check the selector and only `sleep(1)` for the context menu if you see it being an issue!

Reply:
I see, let's leave it be then!